src/mol-plugin-ui/sequence/sequence.tsx

```diff
@@ -320,9 +346,38 @@ export class Sequence<P extends SequenceProps> extends PluginUIComponent<P> {
         this.highlightQueue.next({ seqIdx: -1, buttons, button, modifiers });
     };
 
+    /** This is to avoid infinite loop of attaching custom properties and forcing update */
+    private attachedCustomPropertiesFor?: { theme: ColorThemeProvider, structure: Structure };
+    private updateCustomColorFunction() {
+        const experimentalSequenceColorTheme: BehaviorSubject<ColorThemeProvider> | undefined = this.plugin.customUIState.experimentalSequenceColorTheme;
+        if (!experimentalSequenceColorTheme) return;
+
+        const theme = experimentalSequenceColorTheme.value;
+        if (theme) {
+            const structure = this.props.sequenceWrapper.getLoci(0)?.structure;
+            const themeContext: ThemeDataContext = { structure };
+            this.customColorFunction = theme.factory(themeContext, theme.defaultValues).color;
+            if (theme.ensureCustomProperties) {
+                const isAttached = this.attachedCustomPropertiesFor?.theme === theme && this.attachedCustomPropertiesFor?.structure === structure;
+                if (!isAttached) {
+                    console.log('ensuring', structure.model.entryId, structure.elementCount)
+                    this.attachedCustomPropertiesFor = { theme, structure };
+                    this.plugin.runTask(Task.create('Ensure custom properties for coloring theme', async runtime => {
+                        await theme.ensureCustomProperties?.attach({ assetManager: this.plugin.managers.asset, runtime }, themeContext);
+                        this.forceUpdate();
+                    }));
+                }
+            }
+        } else {
+            this.customColorFunction = undefined;
+        }
+    }
+
     render() {
-        const sw = this.props.sequenceWrapper;
+        console.log('render')
+        this.updateCustomColorFunction();
```

Comment:
Why not create the function just once in `this.subscribe(this.plugin.managers.structure.focus.behaviors.current` and `this.subscribe(experimentalSequenceColorTheme,`?